### PR TITLE
Add details about PATCH behaviour (fixes #566)

### LINKED ISSUE #566: semantics for PATCH should be better specified
See #208, Mozilla-services/cliquet#597.

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #566 

r? @glasserc  (implementation of json merge is left for #623)

### CODE REVIEW

File: docs/api/1.x/_details-patch-object.rst [hunk 1]
```diff
@@ -10,10 +10,23 @@ changed meanwhile, a |status-412| error is returned.
...
 
+
+Attributes merge
+----------------
+
+The provided values are merged with the existing object. For example:
+
+* ``{"a":"b"}`` + ``{"a":"c"}`` → ``{"a":"c"}``
+* ``{"a":"b"}`` + ``{"b":"c"}`` → ``{"a":"b", "b":"c"}``
+* ``{"a":"b"}`` + ``{"a":null}`` → ``{"a":null}`` : *attributes can't be removed with patch*
```
Comment (Natim) [inline docs/api/1.x/_details-patch-object.rst]: That's not the spec, we should remove it

File: docs/api/1.x/_details-patch-object.rst [hunk 2]
```diff
@@ -10,10 +10,23 @@ changed meanwhile, a |status-412| error is returned.
...
+* ``{"a": {"b":"c"}}`` + ``{"a":{"d":"e"}}`` → ``{"a":{"d":"e"}}`` : *sub-objects are replaced, not merged*
```
Comment (Natim) [inline docs/api/1.x/_details-patch-object.rst]: That's not [the spec](https://tools.ietf.org/html/rfc7396)

File: docs/api/1.x/_details-patch-object.rst [hunk 3]
```diff
@@ -10,10 +10,23 @@ changed meanwhile, a |status-412| error is returned.
...
+
 .. note::
 
-    `JSON-Patch <http://jsonpatch.com>`_ is currently not
-    supported. Any help is welcomed though!
+    `JSON-Patch <http://jsonpatch.com>`_ or `JSON merge <https://tools.ietf.org/html/rfc7396>`
```
Comment (Natim) [inline docs/api/1.x/_details-patch-object.rst]: missing _ at the end.

File: docs/api/1.x/_details-patch-object.rst (hunk 1 above)
Comment (Natim) [inline docs/api/1.x/_details-patch-object.rst]: Sorry forget this message, you are describing the current behavior and not the one we want :)

**Comment by Natim** [comment]
f+

**Comment by leplatrem** [comment]
I merged to include it in the next release. If you have feedback I'll take them in another PR

**Pull request merged**